### PR TITLE
(VADC-1228): add access-denied page and flag to enable

### PR DESCRIPTION
**Pull request by ocshawn**
Link to JIRA ticket if there is one: [VADC-1228](https://ctds-planx.atlassian.net/browse/VADC-1228)

### New Features
- add access-denied page and flag to enable when user does not have access to any site resources 

[VADC-1228]: https://ctds-planx.atlassian.net/browse/VADC-1228?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pieterlukasse — COMMENTED** [review]
please rebase and point to https://github.com/uc-cdis/data-portal/tree/fix/rollback_data_dictionary_button if this is a "hotfix" (read: should go to prod soon).

File: src/components/AccessDenied.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      const emailAddressArr = str.match(/[\w\.]+@[\w\.]+/g);
+      if (emailAddressArr.length > 0) {
+        const splitStringOnEmail = str.split(emailAddressArr[0]);
+        return (<React.Fragment>{splitStringOnEmail[0]}<a href={`mailto:${emailAddressArr[0]}`}>{emailAddressArr[0]}</a>{splitStringOnEmail[1]}</React.Fragment>);
+      }
+      return (<React.Fragment>{str}</React.Fragment>);
+    };
+    return (
+      <div className='error-placeholder__error-msg'>
+        <h1>{userAccessToSite.noAccessMessage ? convertEmailsToLink(userAccessToSite.noAccessMessage) : 'Access to this site requires special access.'}</h1>
```
Comment (pieterlukasse) [inline src/components/AccessDenied.jsx:18]:
```suggestion
        <h1>{userAccessToSite.noAccessMessage ? convertEmailsToLink(userAccessToSite.noAccessMessage) : 'Access to this site requires special permission.'}</h1>
```

File: docs/portal_config.md [hunk 2]
```diff
@@ -717,6 +717,11 @@ Below is an example, with inline comments describing what each JSON block config
       "service": "query_page"
     }
   },
+  "userAccessToSite": { // optional: user must have access to a resorces to acess the site
+    "enabled": true,// optional: enable ristricted access
+    "noAccessMessage": "Access to this site requires special access.",// optional: defaults to this value, first email addresses will be turned into mailto link if used
```
Comment (pieterlukasse) [inline docs/portal_config.md:722]:
```suggestion
    "noAccessMessage": "Access to this site requires special permission.",// optional: defaults to this value, first email addresses will be turned into mailto link if used
```

File: src/components/AccessDenied.jsx (hunk 1 above)
Comment (ocshawn) [reply src/components/AccessDenied.jsx:18]: this was word for word what was requested, but i will make your change

**Comment by ocshawn** [comment]
@pieterlukasse created a separate PR for merging to other branch https://github.com/uc-cdis/data-portal/pull/1564 should this still be merged to master?

**Comment by mfshao** [comment]
added a few comments in https://github.com/uc-cdis/data-portal/pull/1564 that maybe should get addressed in this PR

**Comment by ocshawn** [comment]
do not merge unless you incorporate changes from other branch

**Comment by pieterlukasse** [comment]
@ocshawn at some point we should take all the work that is being merged in the other target branch and merge that to `master`. Let's wait for the Data Dictionary feature to be back in scope.

**Comment by ocshawn** [comment]
closing as code will go with https://github.com/uc-cdis/data-portal/pull/1573

**Pull request closed without merging**